botocore/parsers.py

```diff
@@ -507,18 +507,18 @@ def _handle_timestamp(self, shape, value):
         return self._timestamp_parser(value)
 
     def _do_error_parse(self, response, shape):
-        body = json.loads(response['body'].decode(self.DEFAULT_ENCODING))
-        error = {"Error": {}, "ResponseMetadata": {}}
+        body = self._parse_body_as_json(response['body'])
+        error = {"Error": {"Message": '', "Code": ''}, "ResponseMetadata": {}}
```

Comment:
Ah yes I remember that PR.  I think the common theme is that I'd prefer to try to fix failures at their root cause rather than fixing the "effects" of those failure.  In that case, #677 was fixing the the symptom of a bug rather than the "real" root cause, which ended up being https://github.com/boto/botocore/pull/690.

With the `Code/Message` issue, I remember that rather than just have the Exception class handle missing `code/messages`, I wanted to update our parsers/serializers to ensure we always populate code/message, or in this specific case to use the missing `__type` key we weren't looking at.

Would it be fair to say that this isn't the same issue here?